src/libPMacc/include/types.h

```diff
@@ -60,6 +61,16 @@ typedef long long int int64_cu;
 #define DINLINE __device__ __forceinline__
 #define HINLINE __host__ inline
 
+/**
+ * CUDA architecture version (aka PTX ISA level)
+ * 0 for host compilation
+ */
+#ifndef __CUDA_ARCH__
```

Comment:
This would not work in this case. I'd need to use __CUDA_ARCH__ in the code itself and the compiler complains about it beeing not defined (for some reason, device only code gets also evaluated on host)
Using a work-around would not be helpful, so I'd introduce it here (as its the prober way) and fix usages in a separate PR if wanted